src/sync.cpp

```diff
@@ -60,6 +60,11 @@ struct CLockLocation {
             mutexName, sourceFile, itostr(sourceLine), (fTry ? " (TRY)" : ""), m_thread_name);
     }
 
+    std::string Name() const
```

Comment:
Maybe change to `const std::string& Name()` so that it will only copy/allocate it if needed?

Reply:
#17198

Reply:
oh god. thanks.